packages/bundler-plugin-core/src/sentry/releasePipeline.ts

```diff
@@ -127,12 +127,29 @@ export async function setCommits(options: InternalOptions, ctx: BuildContext): P
   span?.finish();
 }
 
-export async function addDeploy(
-  /* version: string, */
-  ctx: BuildContext
-): Promise<string> {
-  const span = addSpanToTransaction(ctx, "function.plugin.add_deploy");
+export async function addDeploy(options: InternalOptions, ctx: BuildContext): Promise<void> {
+  const span = addSpanToTransaction(ctx, "function.plugin.deploy");
```

Comment:
l: I guess this is just taken from the previous/current implementation, but does it make sense to create this span even when this is disabled, or would it not make more sense to skip creating this span in that case? Just out of interest :)

Reply:
I was also already thinking about this (for all optional steps such as `cleanArtifacts`, `addDeploy` or `setCommits`). I think we have two options: 
* skip the span
* set a different span status (e.g. `cancelled`) if the option is not selected

I've extracted #99 from #50 to track everything we need to revisit in terms of the `telemetry` option

Reply:
👍 personally I'd go with skipping them, as they are basically 0-length spans with little information, and you can infer by their absence/presence if they have been skipped or not. but setting a span status is for sure also OK :)